haystack/nodes/prompt/prompt_node.py

```diff
@@ -911,11 +957,13 @@ def run(
         if meta and "meta" not in invocation_context.keys():
             invocation_context["meta"] = meta
 
-        results = self(prompt_collector=prompt_collector, **invocation_context)
+        if "prompt_template" not in invocation_context.keys():
+            invocation_context["prompt_template"] = self.get_prompt_template(prompt_template)
```

Comment:
If someone before set the `prompt_template` to the invocation context, use this. Otherwise resolve it from the param.